themes/src/main/resources/theme/base/login/resources/js/authChecker.js

```diff
@@ -22,21 +27,35 @@ export function checkCookiesAndSetTimer(loginRestartUrl) {
   const session = getSession();
 
   if (!session) {
-    // The session is not present, check again later.
+    // No new session detected, check again later.
     timeout = setTimeout(
-      () => checkCookiesAndSetTimer(loginRestartUrl),
-      CHECK_INTERVAL_MILLISECS,
+      () => startSessionPolling(redirectUrl),
+      SESSION_POLLING_INTERVAL,
     );
   } else {
-    // Redirect to the login restart URL. This can typically automatically login user due the SSO
-    location.href = loginRestartUrl;
+    // A new session was detected, redirect to the specified URL and stop polling.
+    location.href = redirectUrl;
+    stopSessionPolling();
```

Comment:
Is this line executed?

Reply:
Yes, starting a navigation does not prevent additional JavaScript from executing whilst the current page is being unloaded.